File: leetcode/python/884.uncommon-words-from-two-sentences.py
class Solution:
    def uncommonFromSentences(self, s1: str, s2: str) -> List[str]:
        counts = defaultdict(int)
        for word in s1.split(" "):
            counts[word] += 1
        for word in s2.split(" "):
            counts[word] += 1
        return [word for word in counts.keys() if counts[word]==1]

File: leetcode/go/884.uncommon-words-from-two-sentences.go
func uncommonFromSentences(s1 string, s2 string) []string {
    counts := make(map[string]int)
    for _, word := range strings.Fields(s1) {
        if _, ok := counts[word]; ok {
            counts[word]++
        } else {
            counts[word] = 1
        }
    }
    for _, word := range strings.Fields(s2) {
        if _, ok := counts[word]; ok {
            counts[word]++
        } else {
            counts[word] = 1
        }
    }
    ans := make([]string, 0, 100)
    for word, count := range counts {
        if count == 1 {
            ans = append(ans, word)
        }
    }
    return ans
}

